src/main/trial.ts

```diff
@@ -0,0 +1,115 @@
+import { Logger } from '@nodescript/logger';
+import { Redis } from 'ioredis';
+import { config } from 'mesh-config';
+import { dep } from 'mesh-ioc';
+
+import { AccessForbidden } from './ac-auth.js';
+
+export interface TrialServiceRestriction {
+    serviceName: string;
+    requestCount: number;
+}
+
+export interface Trial {
+    serviceRestrictions: Array<TrialServiceRestriction>;
+}
+
+export interface TokenServiceRestriction {
+    serviceName: string;
+    requestLimit: number;
+}
+
+export class TrialClient {
+
+    @config() private REDIS_URL!: string;
+    @config({ default: 'cache:framework:trialClient' }) private TRIAL_KEY_PREFIX!: string;
```

Comment:
I'd say this shouldn't be configurable because we need it to be the same across the services and the backend we will build